src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1000,603 +946,315 @@ private ImmutableArray<Symbol> CreateMembers()
+                        typesBuilder.AddRange(tupleCompatibleType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics, RestPosition - 1);
+                        typesBuilder.AddRange(extensionTupleElementTypes);
+                        elementTypes = typesBuilder.ToImmutableAndFree();
+                    }
+                    else
+                    {
+                        elementTypes = tuple.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+                    }
 
-        public override Accessibility DeclaredAccessibility
-        {
-            get
-            {
-                if (_underlyingType.IsErrorType())
-                {
-                    return Accessibility.Public;
+                    return elementTypes;
                 }
-                else
-                {
-                    return _underlyingType.DeclaredAccessibility;
-                }
-            }
-        }
-
-        public override bool IsStatic
-        {
-            get
-            {
-                return false;
-            }
-        }
-
-        public override bool IsAbstract
-        {
-            get
-            {
-                return false;
-            }
-        }
-
-        public override bool IsSealed
-        {
-            get
-            {
-                return true;
-            }
-        }
-
-        public override int Arity
-        {
-            get
-            {
-                return 0;
-            }
-        }
-
-        public override ImmutableArray<TypeParameterSymbol> TypeParameters
-        {
-            get
-            {
-                return ImmutableArray<TypeParameterSymbol>.Empty;
-            }
-        }
-
-        internal override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotationsNoUseSiteDiagnostics
-        {
-            get
-            {
-                return ImmutableArray<TypeWithAnnotations>.Empty;
-            }
-        }
-
-        public override NamedTypeSymbol ConstructedFrom
-        {
-            get
-            {
-                return this;
             }
-        }
 
-        public override bool MightContainExtensionMethods
-        {
-            get
+            public ImmutableArray<FieldSymbol> TupleElements(NamedTypeSymbol tuple)
             {
-                return false;
-            }
-        }
+                if (_lazyDefaultElementFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyDefaultElementFields, CollectTupleElementFields(tuple));
+                }
 
-        public override string Name
-        {
-            get
-            {
-                return string.Empty;
+                return _lazyDefaultElementFields;
             }
-        }
 
-        internal override bool MangleName
-        {
-            get
+            private ImmutableArray<FieldSymbol> CollectTupleElementFields(NamedTypeSymbol tuple)
```

Comment:
>private [](start = 12, length = 7)

`static`

Reply:
`TupleElementTypesWithAnnotations` uses instance fields

---
In reply to: [355732945](https://github.com/dotnet/roslyn/pull/39370#discussion_r355732945) [](ancestors = 355732945)